### PR TITLE
ZNP-22: creating primitives components for home screen

### PR DESCRIPTION
**Pull request by RogerPino**
Components cardHomeOptionVertical, cardHomeOptionHorizontal, UserFloatCardInfo and ButtonSocialMedia been implemented.
I had to unify the components folder and correct the necessary paths for testing.

### CODE REVIEW

File: src/components/Card/CardHomeOption/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  icon?: JsxElement;
+}
+export default function CardHomeOption({ onClick, text, icon }: TCardHomeOption) {
+  return (
+    <Center boxShadow="md" bg="white" h="200px" w="160px" borderRadius="xl" flexDirection="column" p="5">
+      <Icon color="cyan.500" marginBottom="20px" flex="3" fontSize="7xl" as={BsPersonSquare} />
+      <Flex flex="1" flexDirection="column" >
+        <Text color="gray.500" fontSize="sm">Directorio</Text>
+        <Heading color="cyan.500" fontSize="xl">Pacientes</Heading>
+      </Flex>
```
Comment (acnicolasdc) [inline src/components/Card/CardHomeOption/index.tsx]: Why is the VStack component not used to put the components Vertical?

File: src/components/User/UserFloatCardInfo/index.tsx [hunk 2]
```diff
@@ -18,29 +20,27 @@ type TUserFloatCardInfo = {
...
-            </Box>
+            <Flex pos="relative" height="160px" bg="blue.500" borderTopRadius="xl" justifyContent="center">
+                <Text color="white" fontWeight="bold" fontSize="30px" marginY="20px" marginBottom="50px" >Perfil</Text>
+                <Circle pos="absolute" bottom="-70" bg="cyan.200" borderColor="white" borderWidth="medium" overflow="hidden">
+                    <Image src={user} color="cyan.500" marginBottom="0" />
+                </Circle>
+                <Circle boxShadow="md" size="35px" bg="white" pos="absolute" bottom="-65" right="125">
+                    <Icon fontSize="25px" color="cyan.400" as={FiPlus} />
+                </Circle>
+            </Flex>
```
Comment (acnicolasdc) [inline src/components/User/UserFloatCardInfo/index.tsx:31]: AS I said in other comment, this should be a different component

File: src/components/User/UserFloatCardInfo/index.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
-import { IoIosPerson } from "react-icons/io";
+import { Box, Flex, Circle, Icon, Center, Text, Square, Button, Heading, Image } from "@chakra-ui/react";
 import { FiInstagram } from "react-icons/fi"
 import { CgTwitter } from "react-icons/cg"
 import { FaFacebookF } from "react-icons/fa"
 import { FiPlus } from "react-icons/fi";
 import { SiGooglemaps } from "react-icons/si";
+import user from 'assets/icons/user.svg';
+import ButtonSocialMedia from "../../../components/Button/ButtonSocialMedia"
+
```
Comment (acnicolasdc) [inline src/components/User/UserFloatCardInfo/index.tsx:11]: Ok, man we have a absolute paths... you can just type `import ButtonSocialMedia from "components/Button/ButtonSocialMedia"`

Comment (acnicolasdc) [inline src/components/User/UserFloatCardInfo/index.tsx:11]: Ok, man we have a absolute paths... you can just type `import ButtonSocialMedia from "components/Button/ButtonSocialMedia"`

File: src/components/User/UserFloatCardInfo/index.tsx (hunk 2 above)
Comment (acnicolasdc) [inline src/components/User/UserFloatCardInfo/index.tsx:31]: AS I said in other comment, this should be a different component

File: src/components/User/UserFloatCardInfo/index.tsx (hunk 3 above)
Comment (acnicolasdc) [inline src/components/User/UserFloatCardInfo/index.tsx:11]: Ok, man we have a absolute paths... you can just type `import ButtonSocialMedia from "components/Button/ButtonSocialMedia"`

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { JsxElement } from 'typescript';
```
Comment (acnicolasdc) [inline src/components/Button/ButtonSocialMedia/index.tsx:2]: remove it

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { FiInstagram } from "react-icons/fi"
+import { CgTwitter } from "react-icons/cg"
+import { FaFacebookF } from "react-icons/fa"
+
+
+type TButtonSocialMedia = {
+    onClick?: () => void;
+    text?: String;
+    icon?: String;
+    socialMedia?: String[]
```
Comment (acnicolasdc) [inline src/components/Button/ButtonSocialMedia/index.tsx:13]: remove it

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import { JsxElement } from 'typescript';
+import { Icon, Square } from "@chakra-ui/react";
+import { FiInstagram } from "react-icons/fi"
+import { CgTwitter } from "react-icons/cg"
+import { FaFacebookF } from "react-icons/fa"
+
+
+type TButtonSocialMedia = {
+    onClick?: () => void;
+    text?: String;
```
Comment (acnicolasdc) [inline src/components/Button/ButtonSocialMedia/index.tsx:11]: remove them

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    socialMedia?: String[]
+}
+
+const socialMedia = {
+    Facebook: <Icon fontSize="35px" color="white" p="5px" as={FaFacebookF} />,
+    Instagram: <Icon fontSize="35px" color="white" p="5px" as={FiInstagram} />,
+    Twitter: <Icon fontSize="35px" color="white" p="5px" as={CgTwitter} />
+}
+
+export default function ButtonSocialMedia({ onClick, text, icon = "Facebook" }: TButtonSocialMedia) {
```
Comment (acnicolasdc) [inline src/components/Button/ButtonSocialMedia/index.tsx:22]: remove text and onClick....

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { JsxElement } from 'typescript';
+import { Icon, Square } from "@chakra-ui/react";
+import { FiInstagram } from "react-icons/fi"
+import { CgTwitter } from "react-icons/cg"
+import { FaFacebookF } from "react-icons/fa"
+
+
```
Comment (acnicolasdc) [inline src/components/Button/ButtonSocialMedia/index.tsx:8]: create a type for socialMedia 

`type TSocialMediaOptions = {
Instagram: JsxElement,
......
Facebook: JsxElement,
}`

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+type TButtonSocialMedia = {
+    onClick?: () => void;
+    text?: String;
+    icon?: String;
```
Comment (acnicolasdc) [inline src/components/Button/ButtonSocialMedia/index.tsx:12]: Create a Enum and assigned it, 
 `export enum EIconSocialMedia {
  Facebook,
  Instagram,
  Twitter,
} `

`icon: EIconSocialMedia;`

File: src/components/Card/CardHomeOption/index.tsx (hunk 1 above)
Comment (RogerPino) [reply src/components/Card/CardHomeOption/index.tsx]: Implemented...

File: src/components/User/UserFloatCardInfo/index.tsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+                <Center flexDir="column" position="relative" top="70px" height="250px">
+                    <Heading color="blue.500" fontSize="larger">Dr. Gustavo Martinez</Heading>
+                    <Heading marginY="2px" color="cyan.400" fontSize="larger">Traumatólogo</Heading>
+                    <Heading marginY="2px" color="cyan.400" fontSize="md">Subespecialidad: Neurologia</Heading>
+                    <Heading marginY="2px" color="cyan.400" fontSize="md"> <Icon fontSize="17px" color="cyan.400" as={SiGooglemaps} />Cali, Colombia</Heading>
+                    <Center marginY="5px" h="50px" w="150px" justifyContent="space-between" position="relative">
+                        <Square p="2" bg="blue.500"><Icon fontSize="20px" color="white" as={FiInstagram} /></Square>
+                        <Square p="2" bg="blue.500"><Icon fontSize="20px" color="white" as={CgTwitter} /></Square>
+                        <Square p="2" bg="blue.500"><Icon fontSize="20px" color="white" as={FaFacebookF} /></Square>
+                    </Center>
```
[the remark this answers is not in the document]
Comment (RogerPino) [reply src/components/User/UserFloatCardInfo/index.tsx]: Done and implemented

File: src/providers/chakra/index.ts [hunk 11]
```diff
@@ -36,3 +37,35 @@ export const theme = extendTheme(
...
+      gray: {
+        100: '#F8FCFE',
+        200: '#E1F1F3',
+      },
+      pink: {
+        500: '#E72366',
+      },
+      purple: {
+        500: '#8B2F89',
+      },
```
[the remark this answers is not in the document]
Comment (RogerPino) [reply src/providers/chakra/index.ts]: Checked

File: src/pages/testPage.tsx [hunk 12]
```diff
@@ -16,10 +16,18 @@ export default function Test() {
...
           <UserFloatCardInfo></UserFloatCardInfo>
           <Link color="teal.500" href="https://chakra-ui.com" fontSize="2xl" target="_blank" rel="noopener noreferrer">
             Learn Chakra
+=======
+          <CardMedicalDiagnosis />
+          <CardScheduledAppointments />
+          <CardMedicalPrescription />
+          <Link color="teal.500" href="https://github.com/acnicolasdc/zanar-prescriptor-ui" fontSize="2xl" target="_blank" rel="noopener noreferrer">
+            Current Github repo
+>>>>>>> bcd7d1cffe9cb5072517e5e2cd08247c59d399bf
```
[the remark this answers is not in the document]
Comment (RogerPino) [reply src/pages/testPage.tsx]: Checked

File: src/components/User/UserFloatCardInfo/index.tsx (hunk 2 above)
Comment (RogerPino) [reply src/components/User/UserFloatCardInfo/index.tsx]: repeated ?

File: src/components/User/UserFloatCardInfo/index.tsx (hunk 3 above)
Comment (RogerPino) [reply src/components/User/UserFloatCardInfo/index.tsx]: Sorry... will not happen again

File: src/components/User/UserFloatCardInfo/index.tsx (hunk 2 above)
Comment (RogerPino) [reply src/components/User/UserFloatCardInfo/index.tsx]: repeated true ?

File: src/components/User/UserFloatCardInfo/index.tsx (hunk 3 above)
Comment (RogerPino) [reply src/components/User/UserFloatCardInfo/index.tsx]: I got it

Comment (RogerPino) [reply src/components/User/UserFloatCardInfo/index.tsx]: resolved

File: src/components/Button/ButtonSocialMedia/index.tsx (hunk 4 above)
Comment (RogerPino) [reply src/components/Button/ButtonSocialMedia/index.tsx]: Removed

File: src/components/Button/ButtonSocialMedia/index.tsx (hunk 6 above)
Comment (RogerPino) [reply src/components/Button/ButtonSocialMedia/index.tsx]: Done

File: src/components/Button/ButtonSocialMedia/index.tsx (hunk 8 above)
Comment (RogerPino) [reply src/components/Button/ButtonSocialMedia/index.tsx]: Done

File: src/components/Button/ButtonSocialMedia/index.tsx [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    socialMedia?: String[]
+}
+
+const socialMedia = {
```
[the remark this answers is not in the document]
Comment (RogerPino) [reply src/components/Button/ButtonSocialMedia/index.tsx]: Done

File: src/components/Button/ButtonSocialMedia/index.tsx (hunk 9 above)
Comment (RogerPino) [reply src/components/Button/ButtonSocialMedia/index.tsx]: Done.. thanks

File: src/components/Card/CardHomeOptionVertical/index.tsx [hunk 14]
```diff
@@ -14,10 +12,10 @@ export default function CardHomeOption({ onClick, text, icon }: TCardHomeOption)
   return (
     <Center boxShadow="md" bg="white" h="200px" w="160px" borderRadius="xl" flexDirection="column" p="5">
       <Image src={schedule} color="cyan.500" marginBottom="20px" flex="3" fontSize="7xl" />
-      <Flex flex="1" flexDirection="column" >
+      <VStack flex="1" flexDirection="column" >
```
Comment (acnicolasdc) [inline src/components/Card/CardHomeOptionVertical/index.tsx:15]: VStack doesn't needs flexDirection property, because V === Column

Comment (RogerPino) [reply src/components/Card/CardHomeOptionVertical/index.tsx:15]: Ohhh, sorry...

**Review by acnicolasdc — APPROVED** [review]

**Pull request merged**